### DISCUSSION TITLE
remove Image in  ImageSelector

### DISCUSSION THREAD

**Discussion by md131376st** [HandyOrg/HandyControl#835] [Q&A] [answered]
Hi 
I was wondering how I can remove Image in the ImageSelector in c# code behind or bind the removed command on a custom  button in user Control

**Comment by mhhg** [comment]
`<Button Command="interactivity:ControlCommands.Switch" ...>`
[ImageSelectorBaseStyle.xaml](https://github.com/HandyOrg/HandyControl/blob/0762db0fb4b4a7378d8b8b783d09d30e7c306092/src/Shared/HandyControl_Shared/Themes/Styles/Base/ImageSelectorBaseStyle.xaml#L28)


`public ImageSelector() => CommandBindings.Add(new CommandBinding(ControlCommands.Switch, SwitchImage));`
[ImageSelector.cs](https://github.com/HandyOrg/HandyControl/blob/0762db0fb4b4a7378d8b8b783d09d30e7c306092/src/Shared/HandyControl_Shared/Controls/Input/ImageSelector.cs#L15
)

[Call command from code behind](https://stackoverflow.com/questions/10126968/call-command-from-code-behind)

**Comment by ghost1372** [comment]
Naturally, there is no way, Uri property is readonly. I will solve this problem soon
until then, You can try @mhhg solution

**Comment by mhhg** [comment, marked as answer]
@ghost1372  I just made `DependencyProperty` registeration of `Uri`, `HasValue`, `PreviewBrush` as normal, instead of being read only on the `ImageSelector` class, and created a [pull request #836](https://github.com/HandyOrg/HandyControl/pull/836).

@md131376st I updated the HandyControl demo on a [specific demo branch](https://github.com/mhhg/HandyControl/tree/demo).
You can add an external element to control `ImageSelector` [as follows](https://github.com/mhhg/HandyControl/commit/d504b8a086777b4878effe2bf18907a11d02f292):
1. Add a remove `Button` with a Name attribute and also assign a Name to the `ImageSelector`.
2. Add an event handler method for `OnClick` event of the `Button` on the C# class
3. Set `Uri`, `HasValue`, `PreviewBrush` of the specified `ImageSelector`.

https://user-images.githubusercontent.com/10318906/119201396-1387c580-baa4-11eb-991e-b53b5ad2d5ca.mp4
